src/workers/opportunity/storeCandidateApplicationScore.ts

```diff
@@ -0,0 +1,53 @@
+import { TypedWorker } from '../worker';
+import { TypeORMQueryFailedError } from '../../errors';
+import { ApplicationScored } from '@dailydotdev/schema';
+import { OpportunityMatch } from '../../entity/OpportunityMatch';
+import { applicationScoreSchema } from '../../common/schema/opportunities';
+
+export const storeCandidateApplicationScore: TypedWorker<'gondul.v1.candidate-application-scored'> =
+  {
+    subscription: 'api.store-candidate-application-score',
+    handler: async ({ data }, con, logger): Promise<void> => {
+      try {
+        const { userId, opportunityId, score, description } = data;
+        if (!userId || !opportunityId) {
+          logger.warn(
+            { data },
+            'Missing userId or opportunityId in candidate application score',
+          );
+          return;
+        }
+
+        const applicationRank = applicationScoreSchema.parse({
+          score,
+          description,
+        });
+
+        await con.transaction(async (manager) => {
+          await manager.getRepository(OpportunityMatch).upsert(
+            {
+              userId,
+              opportunityId,
+              applicationRank,
+            },
+            {
+              conflictPaths: ['userId', 'opportunityId'],
+              skipUpdateIfNoValuesChanged: true,
+            },
+          );
+        });
+      } catch (originalError) {
+        const err = originalError as TypeORMQueryFailedError;
+        if (err?.name === 'QueryFailedError') {
+          logger.error({ err, data }, 'could not update application score');
+
+          return;
+        }
+
+        throw err;
+      }
+    },
+    parseMessage: (message) => {
+      return ApplicationScored.fromBinary(message.data);
+    },
```

Comment:
```suggestion
    parseMessage: (message) => ApplicationScored.fromBinary(message.data),
```

Reply:
Also maybe we can start putting this further up? That way it's easier to see and not forget about it :lolsob:

Reply:
<img width="790" height="315" alt="image" src="https://github.com/user-attachments/assets/632b0fcd-a17b-42bc-9454-4edf2420c107" />